src/main.rs

```diff
@@ -167,8 +176,25 @@ impl Args {
             .value_of_lossy("votes")
             .unwrap()
             .parse()?;
+        let (rename_action, dest_dir): (RenameAction, Option<PathBuf>) =
+            if matches.is_present("symlink") {
+                if cfg!(windows) {
+                    failure::bail!(
+                        "symlink not supported on Windows, try hardlink");
+                }
+                (RenameAction::Symlink, Some(matches.value_of_os("symlink")
+                                             .unwrap()
+                                             .to_os_string().into()))
+            } else if matches.is_present("hardlink") {
+                (RenameAction::Hardlink, Some(matches.value_of_os("hardlink")
+                                              .unwrap()
+                                              .to_os_string().into()))
```

Comment:
Do you specifically have a use case for hardlinks? If not, please remove it. Hardlinks are weird and I'd rather not encourage their use.

Reply:
I currently use symlinks because my fuse filesystem doesn't support hardlinks, otherwise I would use hardlinks.

I agree they can be weird, but I think the user should decide the best type of file link for their situation

situation: I want a well organized media directory with sane filenames. Other applications still need access to the original filenames. with hardlinks, there is a cleaner separation between the two uses. If I delete from the media folder, it will still be available for the other application, and vice versa. When neither hardlink exists, the file is deleted.

With symlinks, I only get one side of that indirection. I can't delete from the original location and keep the media folder intact. The file is deleted at once.